docs/adr/0001-extend-tosca-yaml-model.md

```diff
@@ -0,0 +1,45 @@
+# Extend TOSCA YAML model to a full in-memory model
+
+As a programmer, I want to access the complete YAML model and not be forced to traverse through all files.
+
+For instance, `TNodeTemplate.java` offers `public QName getType() {`, but one wants `public TNodeType getType()`.
+
+## Considered Alternatives
+
+* Inherit from plain data model and add transparent resolution
+* New data model
+* Eclipse EMF
+
+## Decision Outcome
+
+* Chosen Alternative: Inherit from plain data model and add transparent resolution
+* Comes out best (see below)
+
+
+## Pros and Cons of the Alternatives <!-- optional -->
+
+### Inherit from plain data model and add transparent resolution
+
+A new model with all classes in a different package inheriting from all model classes.
```

Comment:
I don't think we have to extend the model.
For resolving, we can traverse the merged (after step 1) TServiceTemplate and compare qnames.
And for the plugins, the qname/simple name of the node template's type is sufficient (after nodetypes are resolved)

Reply:
If that is enough, you do not need to touch the model at all. Just implement a lightweight repository, which parses all `.yml` files in a repository and offer a QName resolution.

With the QName resolution, the question is, where to cast to TNodeType / TRelationsipType.

`public Object resolve(QName qname)` or `public TNodeType resolveNodeType(QName qname)` or `public TNodeType resolve(NodeTypeId)`.

In the last case, the caller has to create an appropriate Id. E.g., new NodeTypeId(qname). The caller knows that he wants to resolve a NodeType's QName.

In winery, we do the last option. See: https://github.com/eclipse/winery/blob/1dfac23e558fdc7bcd2122a4d98c89e54d16a3f0/org.eclipse.winery.common/src/main/java/org/eclipse/winery/common/interfaces/IWineryRepositoryCommon.java#L84

`default TNodeType getElement(NodeTypeId id) {`